src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,330 @@
+  gamma::RealT               # ratio of specific heats
+  inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+  Re::RealT                  # Reynolds number
+  Pr::RealT                  # Prandtl number
+  Ma_inf::RealT              # free-stream Mach number
+  kappa::RealT               # thermal diffusivity for Fick's law
+
+  p_inf::RealT               # free-stream pressure
+  u_inf::RealT               # free-stream velocity
+  R::RealT                   # gas constant (depends on nondimensional scaling!)
+
+  equations_hyperbolic::E    # CompressibleEulerEquations2D
+end
+
+function CompressibleNavierStokesEquations2D(equations::CompressibleEulerEquations2D; Reynolds, Prandtl, Mach_freestream)
+  γ = equations.gamma
+  inv_gamma_minus_one = equations.inv_gamma_minus_one
+  Re, Pr, Ma = promote(Reynolds, Prandtl, Mach_freestream)
+
+  # Under the assumption of constant Prandtl number the thermal conductivity
+  # constant is kappa = gamma μ R / ((gamma-1) Pr).
+  # Important note! Due to nondimensional scaling R = 1 / gamma, this constant
+  # simplifies slightly. Also, the factor of μ is accounted for later.
+  κ = inv_gamma_minus_one / Pr
+
+  # From the nondimensionalization discussed above set the remaining free-stream
+  # quantities
+  p_inf = 1 / γ
+  u_inf = Mach_freestream
+  R     = 1 / γ
+  CompressibleNavierStokesEquations2D{typeof(γ),typeof(equations)}(γ, inv_gamma_minus_one,
+                                                                   Re, Pr, Ma, κ,
+                                                                   p_inf, u_inf, R,
+                                                                   equations)
+end
+
+
+# This is the flexibility a user should have to select the different gradient variable types
+# varnames(::typeof(cons2prim)   , ::CompressibleNavierStokesEquations2D) = ("v1", "v2", "T")
+# varnames(::typeof(cons2entropy), ::CompressibleNavierStokesEquations2D) = ("w2", "w3", "w4")
+
+varnames(variable_mapping, equations_parabolic::CompressibleNavierStokesEquations2D) =
+  varnames(variable_mapping, equations_parabolic.equations_hyperbolic)
+
+# we specialize this function to compute gradients of primitive variables instead of
+# conservative variables.
+gradient_variable_transformation(::CompressibleNavierStokesEquations2D, dg_parabolic) = cons2prim
+
+# no orientation specified since the flux is vector-valued
+# Explicit formulas for the diffussive Navier-Stokes fluxes are available, e.g. in Section 2
+# of the paper by Svärd, Carpenter and Nordström
+# "A stable high-order finite difference scheme for the compressible Navier–Stokes
+#  equations, far-field boundary conditions"
+# Although these authors use a different nondimensionalization so some constants are different
+# particularly for Fick's law.
+#
+# Note, could be generalized to use Sutherland's law to get the molecular and thermal
+# diffusivity
```

Comment:
Create an issue? That would be a nice warm-up project for a student.

Reply:
Done in https://github.com/trixi-framework/Trixi.jl/issues/1195